### PR TITLE
fix(test): fix error handling condition in test

### PR DESCRIPTION
**Pull request by Ithrael**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
Fix error handling condition in test

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/apis/podresources/server_v1_test.go [hunk 1]
```diff
@@ -710,7 +710,7 @@ func TestGetPodResourcesV1(t *testing.T) {
 					t.Errorf("want exit = %v, got %v", tc.err, err)
 				}
 			} else {
-				if err != err {
+				if err != nil {
```
Comment (bart0sh) [inline pkg/kubelet/apis/podresources/server_v1_test.go:713]: This condition will never be true it seems as `err != nil` is handled above.

File: pkg/kubelet/apis/podresources/server_v1_test.go [hunk 2]
```diff
@@ -710,12 +710,8 @@ func TestGetPodResourcesV1(t *testing.T) {
 					t.Errorf("want exit = %v, got %v", tc.err, err)
 				}
 			} else {
-				if err != err {
```
Comment (bart0sh) [inline pkg/kubelet/apis/podresources/server_v1_test.go:713]: I guess the original intention was to compare `err` with `tc.err` here. WDYT?

Comment (Ithrael) [reply pkg/kubelet/apis/podresources/server_v1_test.go:713]: It seems that the logic in line 708 with `if err != nil` has already compared err with tc.err, and this logic is inside else, so err is nil, so there is no need to compare err. Just comparing `tc.expectedResponse `with `resp` is enough.

Comment (bart0sh) [reply pkg/kubelet/apis/podresources/server_v1_test.go:713]: There is a need to catch false positives, i.e. situations with expected non-nil error (tc.err) and returned nil error.
Does this make sense to you?

**Comment by Ithrael** [comment]
May I ask what is the progress of this PR?

**Pull request merged**